### DISCUSSION TITLE
What features do you guys want in getashell?

### DISCUSSION THREAD

**Discussion by steveiliop56** [steveiliop56/getashell#7] [Q&A]
Sooo... I have a lot of ideas about features I could add to getashell, some complicated some not. But I don't know what to prioritize. What's a missing feature you guys want me to add?

**Comment by MrSuttonmann** [comment]
An integrated shell UI might be a nice addition, for those who don't want to have to open an ssh client.

**Reply by steveiliop56** [reply]
Aha I see. It would be nice to have a built in ssh session. I will see if I can find a library to do that with React.

**Comment by thagoat** [comment]
Maybe a description of what extra args can be added in the dialog box. Also maybe basic auth for the webui, for those that can't/won't put this behind a reverse proxy if running outside of a LAN.

**Reply by steveiliop56** [reply]
Oh cool, I will add a small help message when you hover over it.

**Reply by steveiliop56** [reply]
Ok I forgot the help message 😅 but I added some basic auth

**Reply by thagoat** [reply]
Awesome!

**Reply by walterzilla** [reply]
> I forgot the help message

Can you at least share here with us the available args? Thanks!

**Reply by steveiliop56** [reply]
The extra args are any normal docker run arguments you would pass to docker, you can add extra ports, extra volumes whatever you want.

**Reply by walterzilla** [reply]
> The extra args are any normal docker run arguments you would pass to docker, you can add extra ports, extra volumes whatever you want.

Oops it seems I totally misunderstand that field scope...

Last question: any chance you could add an option to customize username?

**Reply by steveiliop56** [reply]
Hmm I guess it could be possible but it would be a bit hard since I would have to move the username commands from the Dockerfile to a docker exec command but I can try.
